src/server/enterprise/server/api_server.go

```diff
@@ -200,15 +201,20 @@ func (a *apiServer) Activate(ctx context.Context, req *ec.ActivateRequest) (resp
 	record := &ec.EnterpriseRecord{License: heartbeatResp.License}
 
 	// If the test heartbeat succeeded, write the state and config to etcd
-	if _, err := col.NewSTM(ctx, a.env.EtcdClient, func(stm col.STM) error {
-		if err := a.configCol.ReadWrite(stm).Put(configKey, &ec.EnterpriseConfig{
+	if err := a.env.TxnEnv.WithWriteContext(ctx, func(txCtx *txncontext.TransactionContext) error {
+		if err := a.configCol.ReadWrite(txCtx.SqlTx).Put(configKey, &ec.EnterpriseConfig{
 			LicenseServer: req.LicenseServer,
 			Id:            req.Id,
 			Secret:        req.Secret,
 		}); err != nil {
 			return err
 		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
 
+	if _, err := col.NewSTM(ctx, a.env.EtcdClient, func(stm col.STM) error {
```

Comment:
So we are writing this to postgres, and etcd.  Why does it also need to be in etcd.  What happens if the etcd state is cleared?

Reply:
So here we're writing to a separate collection that's backed by etcd. Reading into the code here I think there may be a bug that I'm thinking through. In general, the enterprise service should heartbeat to the license service to get it's token routinely, and this token is retrieved over an etcd cache. 

Regardless, I think if I make a change to this logic it should be a separate PR